erdos/src/dataflow/stream/loop_stream.rs

```diff
@@ -27,41 +31,39 @@ where
 {
     id: StreamId,
     phantom: PhantomData<D>,
+    graph: Arc<Mutex<InternalGraph>>,
 }
 
 impl<D> LoopStream<D>
 where
     for<'a> D: Data + Deserialize<'a>,
 {
-    pub fn new() -> Self {
+    pub(crate) fn new(graph: Arc<Mutex<InternalGraph>>) -> Self {
         let id = StreamId::new_deterministic();
-        let loop_stream = Self {
+
+        tracing::debug!("Initializing a LoopStream with ID: {}", id,);
+
+        Self {
             id,
             phantom: PhantomData,
-        };
-        default_graph::add_loop_stream(&loop_stream);
-        loop_stream
+            graph: Arc::clone(&graph),
+        }
     }
 
-    pub fn connect_loop(&self, stream: &OperatorStream<D>) {
-        default_graph::connect_loop(self, stream);
-    }
-}
+    pub fn connect_loop(&mut self, stream: &OperatorStream<D>) {
+        Arc::clone(&self.graph)
+            .lock()
+            .unwrap()
+            .connect_loop(self, stream);
 
-impl<D> Default for LoopStream<D>
-where
-    for<'a> D: Data + Deserialize<'a>,
-{
-    fn default() -> Self {
-        Self::new()
+        tracing::debug!(
+            "Connected LoopStream with ID: {} to stream named: {}",
+            self.id,
+            stream.name,
+        );
     }
-}
 
-impl<D> Stream<D> for LoopStream<D>
-where
-    for<'a> D: Data + Deserialize<'a>,
-{
-    fn id(&self) -> StreamId {
+    pub(crate) fn get_id(&self) -> StreamId {
```

Comment:
What convention are we using in the API? I seem to remember everything is `id()` instead of `get_id()`

Reply:
Switching to id() as the style of `get_` is not used elsewhere